generate.sh

```diff
@@ -184,6 +187,9 @@ fi
 
 print_log "Setting up documentation for Download and Preview\n"
 mv $BUILD_DIR/_build/html $ROOT_DIR/../${UNIQUEID}_preview && cd $_/../
+if [[ "$JAVADOC_PATH" != "" ]]; then
+  mv $BASE/temp/$EMAIL/$UNIQUEID/javadoc $BASE/temp/$EMAIL/${UNIQUEID}_preview
```

Comment:
Instead of copying directly to {uniqueid}_preview, instead do this earlier and copy to `_build/html`. and try to create a single `if [ "&JAVADOC_PATH" != "" ]` block.